src/components/data/LocalVariablesContent.js

```diff
@@ -80,17 +110,34 @@ const LocalVariablesContent = ({ workspace, workspace: { workspace: { googleProj
   )(async originalKey => {
     const isList = editType.includes('list')
     const newBaseType = isList ? editType.slice(0, -5) : editType
-
     const parsedValue = isList ? _.map(Utils.convertValue(newBaseType), editValue.split(/,\s*/)) :
       Utils.convertValue(newBaseType, editValue)
 
-    await Ajax().Workspaces.workspace(namespace, name).shallowMergeNewAttributes({ [editKey]: parsedValue })
+    const attributesToDelete = []
+    const attributesToMerge = { [editKey]: parsedValue }
 
     if (editKey !== originalKey) {
-      await Ajax().Workspaces.workspace(namespace, name).deleteAttributes([originalKey])
+      // Edit key differs from original key =>
+      // delete original key and delete original description key, if present
+      attributesToDelete.push(originalKey)
+      attributesToDelete.push(toDescriptionKey(originalKey))
     }
 
+    const editDescriptionKey = toDescriptionKey(editKey)
+    if (editDescription) {
+      // New description or change to existing one =>
+      // merge editDescription
+      attributesToMerge[editDescriptionKey] = editDescription
+    } else {
+      // Change to description resulting in empty description =>
+      // delete description attribute
+      attributesToDelete.push(editDescriptionKey)
+    }
```

Comment:
I wonder if we can avoid the mutations above via something like

```javascript
    const editDescriptionKey = toDescriptionKey(editKey)

    const attributesToMerge = {
      editKey: parsedValue,
      ...(!!editDescription && { editDescriptionKey: editDescription })
    }

    const attributesToDelete = _.flatten(
      editKey !== originalKey && [originalKey, toDescriptionKey(originalKey)],
      !editDescription && [editDescriptionKey]
    )
```

Reply:
Almost - turns out we needed something like this:

```
    const editDescriptionKey = toDescriptionKey(editKey)

    const attributesToMerge = {
      [editKey]: parsedValue,
      ...(!!editDescription && { [editDescriptionKey]: editDescription })
    }

    const attributesToDelete = _.compact(_.flatten([
      editKey !== originalKey && [originalKey, toDescriptionKey(originalKey)],
      !editDescription && [editDescriptionKey]
    ]
    ))
```

Reply:
Why does `attributesToMerge` need to have an array key; i.e. `[editKey]` instead of `editKey`?

Reply:
Without the array key, the key we merge literally becomes 'editKey' instead of the value referenced by editKey:

![image](https://user-images.githubusercontent.com/1266674/146582757-8a6af501-50cb-4661-8c3f-d8067270083d.png)

Reply:
Ah I see.

Reply:
This is exactly how I'd want to see this used, A+